extensions/mssql/src/typings/promise.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Deferred promise
+ */
+export class Deferred<T = void> {
```

Comment:
https://github.com/microsoft/vscode-mssql/blob/main/src/protocol.ts#L31

(and regardless, the typings folder isn't where stuff like this belongs. There should only be .d.ts files in here)

Reply:
anything else other than location that needs to be changed? I picked this definition of Deferred from one of your files in ADS.

Reply:
No, the definition itself was fine. But in case it wasn't clear we already have a Deferred class in here so you should just use that one